vtStorScsi/CommandHandlerScsi.cpp

```diff
@@ -70,68 +70,125 @@ namespace vtStor
+                CdbRegister.Cdb_16.Registers.Register6 = ((InputFields.Command16.Lba >> 24) & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register7 = ((InputFields.Command16.Lba >> 16) & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register8 = ((InputFields.Command16.Lba >> 8) & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register9 = (InputFields.Command16.Lba & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register10 = ((InputFields.Command16.Length >> 24) & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register11 = ((InputFields.Command16.Length >> 16) & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register12 = ((InputFields.Command16.Length >> 8) & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register13 = (InputFields.Command16.Length & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register14 = InputFields.Command16.MiscCdb;
+                CdbRegister.Cdb_16.Registers.Register15 = InputFields.Command16.Control;
+            } break;
+
+            case StorageUtility::Scsi::eFieldFormatting::ATAPASSTHROUGH_16:
+            {
+                CdbRegister.Cdb_16.Registers.Register0 = InputFields.AtaPassThrough16.OpCode;
+                //     7 6 5            4 3 2 1        0
+                // MULTIPLE_COUNT       PROTOCOL     EXTEND
+                CdbRegister.Cdb_16.Registers.Register1 = (((InputFields.AtaPassThrough16.MultipleCount << 5) & 0xE0) 
+                                                        | ((InputFields.AtaPassThrough16.Protocol << 1) & 0x1E)
+                                                        | (InputFields.AtaPassThrough16.Extend & 0x01));
+                //   7 6           5           4          3        2          1 0
+                // OFF_LINE     CK_COND     Reserved    T_DIR   BYT_BLOK    T_LENGTH
+                CdbRegister.Cdb_16.Registers.Register2 = (((InputFields.AtaPassThrough16.Offline << 6) & 0xC0)
+                                                        | ((InputFields.AtaPassThrough16.CkCond << 5) & 0x20)
+                                                        | ((InputFields.AtaPassThrough16.TDir << 3) & 0x08)
+                                                        | ((InputFields.AtaPassThrough16.ByteBlock << 2) & 0x04)
+                                                        | (InputFields.AtaPassThrough16.TLength & 0x03));
+                CdbRegister.Cdb_16.Registers.Register3 = ((InputFields.AtaPassThrough16.Features >> 8) & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register4 = (InputFields.AtaPassThrough16.Features & 0xFF);
+                CdbRegister.Cdb_16.Registers.Register13 = InputFields.AtaPassThrough16.Device;
+                CdbRegister.Cdb_16.Registers.Register14 = InputFields.AtaPassThrough16.Command;
+                CdbRegister.Cdb_16.Registers.Register15 = InputFields.AtaPassThrough16.Control;
+
+                if (1 == InputFields.AtaPassThrough16.Extend)
```

Comment:
I think it is ok here. In the Ata pass through spec, The Extend is only 1 bit (0-1). 1 means to enable extend command.